### DISCUSSION TITLE
Adding custom logging handler to `uvicorn.error` does not work when running with `fastapi run`

### DISCUSSION THREAD

**Discussion by BreezeWhite** [fastapi/fastapi-cli#44] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import logging
from fastapi import FastAPI


class CustomHandler(logging.StreamHandler):
    def emit(self, record):
        print('No way')


my_handler = CustomHandler()
my_handler.setLevel(logging.ERROR)
uvi_err_logger = logging.getLogger('uvicorn.error')
uvi_err_logger.addHandler(my_handler)

app = FastAPI()

@app.get('/test')
async def test():
    raise ValueError('ohoh')  # This should trigger the error message and print out 'No way' in the terminal
```

### Description

When I am using the custom logging handler, inserting the handler under `uvicorn.error`, then run the app with `fastapi dev`, it works fine that prints out "No way" to the terminal. But it fails when running with `fastapi run`. I looked through the code but cannot find any clue about the why the behaviors are different. This problem is very hard to notice about, and it's been two months later when I finally realized that I've not receiving any error messages from our system...

### Operating System

Linux

### Operating System Details

This problem happens on every platform.

### Python Version

3.11.2

**Comment by skaaptjop** [comment]
Try removing the existing log handlers before adding your custom one

```python
uvi_err_logger = logging.getLogger("uvicorn.access")
   if uahandlers := uvi_err_logger.handlers:
      uvi_err_logger.removeHandler(uahandlers[0])
uvi_err_logger.addHandler(my_handler)
```

I do this sort of thing fairly often when I deploy web apps to google cloud in order to reduce the amount of redundant request logging.

**Reply by skaaptjop** [reply, marked as answer]
Also, look at https://github.com/fastapi/fastapi-cli/discussions/15#discussioncomment-9745019
That worked for me although I haven't bothered yet to find out why

**Reply by BreezeWhite** [reply]
The `lifespan` approach does work, though pretty weird 😂

**Reply by keatonLiu** [reply]
> The `lifespan` approach does work, though pretty weird 😂

That's because when you call uvicorn.run(), the logging settings are overwritten if you put the init logic at top of your file.
